### PR TITLE
fix: Fetch repos when unavailable in context

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by dcramer** [comment]
@billyvg any idea how I can get these tests to... test something useful?

I either:

- Fail using ``mount`` due to ``store`` context propagation (and way too many other things)
- Can't get shallow to test enough of the component

File: webapp/pages/RepositoryDetails.jsx [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
     ...AsyncPage.contextTypes,
+    store: PropTypes.object.isRequired,
     repoList: PropTypes.arrayOf(PropTypes.object).isRequired
   };
 
   static childContextTypes = {
     ...AsyncPage.childContextTypes,
     ...RepositoryDetails.contextTypes,
-    repo: PropTypes.object.isRequired
+    store: PropTypes.object.isRequired,
```
Comment (billyvg) [inline webapp/pages/RepositoryDetails.jsx]: Isn't this already in context via `<Provider>`?

Comment (dcramer) [reply webapp/pages/RepositoryDetails.jsx]: this was a hack to work around enzyme yelling

the problem is i can do like ``<Provider store={mockedStore}><ComponentImTesting /></Provider>``, but then I cant pass context to the Component (as enzyme seems to not propagate it correctly).

It also throws up everywhere yelling about any context thats passed thats not explicitly defined in childContextTypes (per the above)

Comment (billyvg) [reply webapp/pages/RepositoryDetails.jsx]: I think you need to define `contextTypes` and `childContextTypes` in the 2nd arg to `mount`. e.g. https://github.com/getsentry/sentry/blob/master/tests/js/setup.js#L44

**Comment by dcramer** [comment]
@billyvg seem good now? in addition to the async rendering test

(i gave up on figuring out the error boundary tests)

**Pull request merged**